Missing spec field causes Controller CrashLoopBackOff

**What happened**:
FederatedSecrets / FederatedConfigmaps takes up to 30 minutes to be visible in
joined clusters. After several minutes some of them are visible,.....after minutes next
ConfigMaps comes (maybe)
You can't rely on the Secrects or ConfigMaps being synchronized at all. Some are not sync


**What you expected to happen**:
new created Federated#### comes within minutes and not hours

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`)
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-08T16:31:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.1", GitCommit:"4485c6f18cee9a5d3c3b4e523bd27972b1b53892", GitTreeState:"clean", BuildDate:"2019-07-18T09:09:21Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```


- KubeFed version
``` 
kubefedctl version: version.Info{Version:"v0.1.0-rc5-dirty", GitCommit:"99be0218bf5ac7d560ec0d7c2cfbfcbb86ba2d61", GitTreeState:"dirty", BuildDate:"2019-08-01T15:41:41Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}

``` 
- Scope of installation (namespaced or cluster)
cluster

- Others

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

found stacktrace in the federated controller

```
E0805 15:40:40.600477       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:277
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283
/go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1561171]

goroutine 3336 [running]:
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1794f60, 0x2d1e8e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/kubefed/pkg/controller/sync.GetOverrideHash(0xc00048c3e8, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:278 +0xa1
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).OverrideVersion(0xc000d1e5b0, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108 +0x32
sigs.k8s.io/kubefed/pkg/controller/sync/version.(*VersionManager).Update(0xc000295300, 0x1cb0fa0, 0xc000d1e5b0, 0x2d59f18, 0x0, 0x0, 0xc001696ae0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147 +0x93
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).UpdateVersions(0xc000d1e5b0, 0x2d59f18, 0x0, 0x0, 0xc001696ae0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125 +0x6f
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).syncToClusters(0xc0001e2e10, 0x1cee020, 0xc000d1e5b0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373 +0x7a7
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).reconcile(0xc0001e2e10, 0xc00024d110, 0x8, 0xc0007d1c80, 0x12, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283 +0x724
sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).worker(0xc0009a2b90)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156 +0xae
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0007724e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007724e0, 0x3b9aca00, 0x0, 0x1, 0xc0016424e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0007724e0, 0x3b9aca00, 0xc0016424e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).Run
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:119 +0xe5
WDFM33957623A:lis-configuration d023280$ (jenkins-shoot--lis--canary) 

```

I changed the controller image to use "canary". Same Problem.

``` 
E0805 15:59:31.889638       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:32.289507       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:32.694443       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:33.199124       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:33.389431       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:33.803807       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:34.289361       1 reflector.go:251] sigs.k8s.io/kubefed/pkg/controller/util/federated_informer.go:437: Failed to watch <nil>: the server does not allow this method on the requested resource
E0805 15:59:34.389791       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:277
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283
/go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1561251]

goroutine 3126 [running]:
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1794f60, 0x2d1e8e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/kubefed/pkg/controller/sync.GetOverrideHash(0xc000335140, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:278 +0xa1
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).OverrideVersion(0xc0002cbc70, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108 +0x32
sigs.k8s.io/kubefed/pkg/controller/sync/version.(*VersionManager).Update(0xc000284780, 0x1cb0fe0, 0xc0002cbc70, 0x2d59f18, 0x0, 0x0, 0xc0027faed0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147 +0x93
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).UpdateVersions(0xc0002cbc70, 0x2d59f18, 0x0, 0x0, 0xc0027faed0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125 +0x6f
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).syncToClusters(0xc000202870, 0x1cee060, 0xc0002cbc70, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373 +0x7a7
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).reconcile(0xc000202870, 0xc0003caf29, 0x5, 0xc0003caf10, 0xf, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283 +0x724
sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).worker(0xc001110be0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156 +0xae
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0008933e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0008933e0, 0x3b9aca00, 0x0, 0x1, 0xc000a43380)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0008933e0, 0x3b9aca00, 0xc000a43380)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).Run
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:119 +0xe5
```

kubectl version
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.0", GitCommit:"641856db18352033a0d96dbc99153fa3b27298e5", GitTreeState:"clean", BuildDate:"2019-03-26T00:05:06Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.5", GitCommit:"0e9fcb426b100a2aea5ed5c25b3d8cfbb01a8acf", GitTreeState:"clean", BuildDate:"2019-08-05T09:13:08Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

and the pods again in CrashLoopBack
``` 
E0806 10:06:33.933851       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:277
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283
/go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1561171]

goroutine 4278 [running]:
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1794f60, 0x2d1e8e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/kubefed/pkg/controller/sync.GetOverrideHash(0xc001236418, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:278 +0xa1
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).OverrideVersion(0xc000580410, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108 +0x32
sigs.k8s.io/kubefed/pkg/controller/sync/version.(*VersionManager).Update(0xc00028b800, 0x1cb0fa0, 0xc000580410, 0x2d59f18, 0x0, 0x0, 0xc0025c60c0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147 +0x93
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).UpdateVersions(0xc000580410, 0x2d59f18, 0x0, 0x0, 0xc0025c60c0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125 +0x6f
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).syncToClusters(0xc0019dd7a0, 0x1cee020, 0xc000580410, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373 +0x7a7
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).reconcile(0xc0019dd7a0, 0xc000fe5cc9, 0x5, 0xc000fe5cb0, 0xf, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283 +0x724
sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).worker(0xc001323180)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156 +0xae
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000269bb0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000269bb0, 0x3b9aca00, 0x0, 0x1, 0xc001c5ecc0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000269bb0, 0x3b9aca00, 0xc001c5ecc0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).Run
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:119 +0xe5

``` 

one of the joined cluster is still 1.15. But the cluster with the federated control plane is 1.14

`unjoin` member cluster with 1.15 helps to sync secrets and configmaps ....but controller pods still crashing after some minutes

``` 
E0806 10:16:01.353859       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:277
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373
/go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283
/go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1561171]

goroutine 3985 [running]:
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1794f60, 0x2d1e8e0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/kubefed/pkg/controller/sync.GetOverrideHash(0xc0006631d8, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:278 +0xa1
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).OverrideVersion(0xc001ac4820, 0x0, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:108 +0x32
sigs.k8s.io/kubefed/pkg/controller/sync/version.(*VersionManager).Update(0xc001140c00, 0x1cb0fa0, 0xc001ac4820, 0x2d59f18, 0x0, 0x0, 0xc001488570, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/version/manager.go:147 +0x93
sigs.k8s.io/kubefed/pkg/controller/sync.(*federatedResource).UpdateVersions(0xc001ac4820, 0x2d59f18, 0x0, 0x0, 0xc001488570, 0x0, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/resource.go:125 +0x6f
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).syncToClusters(0xc0006137a0, 0x1cee020, 0xc001ac4820, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:373 +0x7a7
sigs.k8s.io/kubefed/pkg/controller/sync.(*KubeFedSyncController).reconcile(0xc0006137a0, 0xc000a32c10, 0x9, 0xc0017dc5e0, 0x13, 0x0)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/sync/controller.go:283 +0x724
sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).worker(0xc001114230)
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:156 +0xae
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc001124450)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc001124450, 0x3b9aca00, 0x0, 0x1, 0xc0009141e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc001124450, 0x3b9aca00, 0xc0009141e0)
        /go/src/sigs.k8s.io/kubefed/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/kubefed/pkg/controller/util.(*asyncWorker).Run
        /go/src/sigs.k8s.io/kubefed/pkg/controller/util/worker.go:119 +0xe5
SAP-Laptop:lis-mixer-rest d023280$ (shoot--lis--canary) 

```

--- a/scripts/pre-commit.sh
+++ b/scripts/pre-commit.sh
@@ -144,7 +144,7 @@ echo "Downloading e2e test dependencies"
 ./scripts/download-e2e-binaries.sh
 
 CREATE_INSECURE_REGISTRY=y CONFIGURE_INSECURE_REGISTRY_HOST=y OVERWRITE_KUBECONFIG=y \
-    KIND_TAG="v1.14.2" ./scripts/create-clusters.sh
+    KIND_TAG="v1.15.0" ./scripts/create-clusters.sh
 
 # Initialize list of clusters to join
 join-cluster-list > /dev/null

--- a/test/e2e/ftccontroller.go
+++ b/test/e2e/ftccontroller.go
@@ -108,6 +108,8 @@ var _ = Describe("FTC controller", func() {
 			tl.Fatalf("Error updating target CRD version %q: %v", existingCrd.Spec.Version, err)
 		}
 
+		waitForTargetCrd(f.Logger(), f.KubeConfig(), targetAPIResource.Name, "v2")
+
 		By("Enabling federation of the CRD v2 to make target version of the FederatedTypeConfig updated")
 		needCleanup = false
 		typeConfig = enableResource(f, &targetAPIResource, "v2", needCleanup)
@@ -134,7 +136,7 @@ func enableResource(f framework.KubeFedFramework, targetAPIResource *metav1.APIR
 
 	resources, err := kfenable.GetResources(f.KubeConfig(), enableTypeDirective)
 	if err != nil {
-		tl.Fatalf("Error retrieving resources to enable federation of target type %q: %v", targetAPIResource.Kind, err)
+		tl.Fatalf("Error retrieving resources to enable federation of target type %q version %q: %v", targetAPIResource.Kind, version, err)
 	}
 	typeConfig := resources.TypeConfig
 

